storage/netapp/netapp_e_hostgroup.py

```diff
@@ -0,0 +1,426 @@
+
+
+def delete_hostgroup(module, ssid, group_id, api_url, user, pwd):
+    groups = "storage-systems/%s/host-groups/%s" % (ssid, group_id)
+    url = api_url + groups
+    # TODO: Loop through hosts, do mapping to href, make new list to pass to data
+    try:
+        rc, data = request(url, method='DELETE', headers=HEADERS, url_username=user, url_password=pwd)
+    except Exception:
+        err = get_exception()
+        module.fail_json(msg="Failed to delete host group. Group [%s]. Id [%s]. Error [%s]." % (group_id, ssid
+                                                                                                , str(err)))
+
+    return rc, data
+
+
+def get_hostgroup_id(module, ssid, name, api_url, user, pwd):
+    all_groups = 'storage-systems/%s/host-groups' % ssid
+    url = api_url + all_groups
+    rc, data = request(url, method='GET', headers=HEADERS, url_username=user, url_password=pwd)
+    for hg in data:
+        if hg['name'] == name:
+            return hg['id']
+        else:
+            continue
+
+    module.fail_json(msg="A hostgroup with the name %s could not be found" % name)
+
+
+def get_hosts_in_group(module, ssid, group_name, api_url, user, pwd):
+    all_groups = 'storage-systems/%s/host-groups' % ssid
+    g_url = api_url + all_groups
+    try:
+        g_rc, g_data = request(g_url, method='GET', headers=HEADERS, url_username=user, url_password=pwd)
+    except Exception:
+        err = get_exception()
+        module.fail_json(
+            msg="Failed in first step getting hosts from group. Group: [%s]. Id [%s]. Error [%s]." % (group_name,
+                                                                                                      ssid,
+                                                                                                      str(err)))
+
+    all_hosts = 'storage-systems/%s/hosts' % ssid
+    h_url = api_url + all_hosts
+    try:
+        h_rc, h_data = request(h_url, method='GET', headers=HEADERS, url_username=user, url_password=pwd)
+    except Exception:
+        err = get_exception()
+        module.fail_json(
+            msg="Failed in second step getting hosts from group. Group: [%s]. Id [%s]. Error [%s]." % (
+                group_name,
+                ssid,
+                str(err)))
+
+    hosts_in_group = []
+
+    for hg in g_data:
+        if hg['name'] == group_name:
+            clusterRef = hg['clusterRef']
+
+    for host in h_data:
+        if host['clusterRef'] == clusterRef:
+            hosts_in_group.append(host['name'])
+
+    return hosts_in_group
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=False),
+            new_name=dict(required=False),
+            ssid=dict(required=True),
+            id=dict(required=False),
+            state=dict(required=True, choices=['present', 'absent']),
+            hosts=dict(required=False, type='list'),
+            api_url=dict(required=False),
```

Comment:
This seems to disagree with the module documentation...

```
  api_url:
    required: true
    description:
    - the resolvable endpoint for the web services proxy. Example, http://10.10.10.10:8080/devmgr/v2
```
ditto for api_username, api_password